sql-to-dbsp-compiler/SQL-compiler/src/test/java/org/dbsp/sqlCompiler/compiler/sql/postgres/PostgresFloat8Tests.java

```diff
@@ -0,0 +1,942 @@
+        this.qf("SELECT atanh('infinity'::double)", "out of range");
+        this.qf("SELECT atanh('-infinity'::double)", "out of range");
+    }
+
+    @Test @Ignore("https://github.com/feldera/feldera/issues/1153")
+    public void castToInt() {
+        this.qf("SELECT '32768.6'::double::int2", "out of range");
+        this.qf("SELECT '-9223372036854780000'::float8::int8", "out of range");
+    }
+
+    @Test
+    public void nonPostgresTests() {
+        this.qs("""
+                SELECT ln(1);
+                 ln
+                ----
+                 0
+                (1 row)
+                
+                SELECT ln(1e0::float8);
+                 ln
+                ----
+                 0
+                (1 row)
+                
+                SELECT log10(1);
+                 log10
+                -------
+                 0
+                (1 row)
+                
+                SELECT log10(1e0::float4);
+                 log10
+                -------
+                 0
+                (1 row)
+                
+                SELECT log(2);
+                 log
+                -----
+                 0.693147180559945
+                (1 row)
+                
+                SELECT log(1.0);
+                 log
+                -----
+                 0
+                (1 row)
+                
+                SELECT log(1, 2);
+                 log
+                -----
+                 0
+                (1 row)
+                
+                SELECT log(1.0, 2.0);
+                 log
+                -----
+                 0
+                (1 row)
+                
+                SELECT log(3.0, 4.0);
+                 log
+                -----
+                 0.792481250360578
+                (1 row)
+                
+                SELECT log(7.7, PI);
+                 log
+                -----
+                 1.783145835617836
+                (1 row)
+                
+                SELECT log(64.0, 2.0);
+                 log
+                -----
+                 6.0
+                (1 row)
+                """
+        );
+    }
+
+    @Test
+    public void testLogFail() {
+        this.runtimeFail("select ln(-1)", "Unable to calculate ln for -1", this.getEmptyIOPair());
+        this.runtimeFail("select log10(-1)", "Unable to calculate log10 for -1", this.getEmptyIOPair());
+        this.runtimeFail("select log(-1)", "Unable to calculate ln for -1", this.getEmptyIOPair());
+        this.runtimeFail("select log(0, -1)", "Unable to calculate log(0, -1)", this.getEmptyIOPair());
+        this.runtimeFail("select log(-1, 0)", "Unable to calculate log(-1, 0)", this.getEmptyIOPair());
+        this.runtimeFail("select log(-1, -1)", "Unable to calculate log(-1, -1)", this.getEmptyIOPair());
+    }
+
+    // Calcite optimized version returns 0
+    @Test @Ignore("https://github.com/feldera/feldera/issues/1348")
```

Comment:
Can we fix this test?
I think we agreed that log in base 0 does return 0.

Reply:
Yes, this is fixed.